config/functions: next wave of cleanups

antonlacon opened the pull request:
This series reorganizes and does cleanup on config/functions. Specifically it:

Adds a die() function, as previously seen in the download-cleaner PR.

Reorganizes all functions into groups based on what they do, and when they are referenced. Functions used by other functions -> buildsystem -> package -> kernel package -> addon package -> target configuration. This commit makes no function changes, only location in the file.

Converts which to command -v, and exit calls to die().

Splits check_config() into its individual checks, adds a check_distro(), and does a cleanup pass.

Built without incident for RPi2.

MilhouseVH commented on config/functions:
```diff
@@ -484,34 +484,57 @@ save_build_config() {
+      distro_err_msg="$distro_err_msg\n - ${distros##*/}"
     done
-    echo -e $check_project
+    echo -e $distro_err_msg
```
If you add `-e` support to die, then can't you combine this so that it becomes `die "$distro_err_msg"`?

antonlacon commented:
Revision pushed. Updated die to use echo -e and to redirect to stderr. Fixed some missing quotes in check_distro. Added a cleanup pass on check_path and check_arch.
